jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialHelper.java

```diff
@@ -113,16 +113,17 @@ public Credential retrieve()
       return retrieve(Arrays.asList(credentialHelper.toString(), "get"));
     }
 
-    try {
-      return retrieve(Arrays.asList(credentialHelper.toString() + ".cmd", "get"));
-
-    } catch (CredentialHelperNotFoundException ex) {
+    // We are on Windows.
+    for (String suffix : Arrays.asList(".cmd", ".exe")) {
       try {
-        return retrieve(Arrays.asList(credentialHelper.toString(), "get"));
+        return retrieve(Arrays.asList(credentialHelper.toString() + suffix, "get"));
       } catch (CredentialHelperNotFoundException ignored) {
-        throw ex;
+        // ignored
       }
     }
+    // On Windows, launching a process from Java without a file extension should normally fail, but
```

Comment:
Do we have a citation for this? Just curious would be helpful to include.

Reply:
Added [this link](https://github.com/GoogleContainerTools/jib/issues/2399#issuecomment-612972912) to the code comment.